charm/kafka/reactive/kafka.py

```diff
@@ -57,6 +46,17 @@ def waiting_for_zookeeper_ready(zk):
     hookenv.status_set('waiting', 'waiting for zookeeper to become ready')
 
 
+@hook('upgrade-charm')
+def upgrade_charm():
+    remove_state('kafka.nrpe_helper.installed')
+    remove_state('kafka.started')
+
+
+@hook('config-changed')
+def config_changed():
+    remove_state('kafka.zk.disabled')
```

Comment:
We'll want to re-disable the embedded zookeeper on upgrade-charm, because the snap may have been reinstalled and that might re-enable the service. However, I don't think we need to do this on config-changed.